Better error message on giveBlock command for non-existing blocks

Pretty much the fix for #2174 in PR #2252, but also applied to the `giveBlocks` command - first fix only seemed to make `giveItem` behave better :-)

Example - enter `giveBlock eastereggdetector` results in the big messy stacktrace below. The EasterEggDetector from the BlockDetector module is actually an item, not a block. Maybe the help text could be smart enough to suggest if the player meant an item when they requested a block? Or should we just make both commands consider *both* items and blocks?

```
23:08:15.771 [main] ERROR o.t.logic.console.ConsoleImpl - An error occurred while executing a command
org.terasology.logic.console.commandSystem.exceptions.CommandExecutionException: java.lang.IllegalArgumentException: No block found for 'eastereggdetector'
	at org.terasology.logic.console.commandSystem.AbstractCommand.execute(AbstractCommand.java:242) ~[classes/:na]
	at org.terasology.logic.console.ConsoleImpl.execute(ConsoleImpl.java:232) [classes/:na]
	at org.terasology.logic.console.ConsoleImpl.execute(ConsoleImpl.java:196) [classes/:na]
	at org.terasology.logic.console.ui.ConsoleScreen.lambda$initialise$89(ConsoleScreen.java:73) [classes/:na]
	at org.terasology.logic.console.ui.ConsoleScreen$$Lambda$243/1523941008.onActivated(Unknown Source) [classes/:na]
	at org.terasology.rendering.nui.widgets.UIText.onKeyEvent(UIText.java:318) [classes/:na]
	at org.terasology.logic.console.ui.UICommandEntry.onKeyEvent(UICommandEntry.java:83) [classes/:na]
	at org.terasology.rendering.nui.internal.NUIManagerInternal.keyEvent(NUIManagerInternal.java:512) [classes/:na]
	at org.terasology.rendering.nui.internal.NUIManagerInternalMethodAccess.invoke(Unknown Source) [reflectasm-1.11.0.jar:na]
	at org.terasology.entitySystem.event.internal.EventSystemImpl$ByteCodeEventHandlerInfo.invoke(EventSystemImpl.java:506) [classes/:na]
	at org.terasology.entitySystem.event.internal.EventSystemImpl.sendConsumableEvent(EventSystemImpl.java:279) [classes/:na]
	at org.terasology.entitySystem.event.internal.EventSystemImpl.send(EventSystemImpl.java:258) [classes/:na]
	at org.terasology.entitySystem.entity.internal.BaseEntityRef.send(BaseEntityRef.java:145) [classes/:na]
	at org.terasology.input.InputSystem.sendKeyEvent(InputSystem.java:475) [classes/:na]
	at org.terasology.input.InputSystem.processKeyboardInput(InputSystem.java:425) [classes/:na]
	at org.terasology.input.InputSystem.update(InputSystem.java:272) [classes/:na]
	at org.terasology.engine.modes.StateIngame.handleInput(StateIngame.java:195) [classes/:na]
	at org.terasology.engine.subsystem.lwjgl.LwjglInput.postUpdate(LwjglInput.java:54) [classes/:na]
	at org.terasology.engine.TerasologyEngine.mainLoop(TerasologyEngine.java:423) [classes/:na]
	at org.terasology.engine.TerasologyEngine.run(TerasologyEngine.java:369) [classes/:na]
	at org.terasology.engine.Terasology.main(Terasology.java:154) [classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_45]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_45]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_45]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_45]
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144) [idea_rt.jar:na]
Caused by: java.lang.IllegalArgumentException: No block found for 'eastereggdetector'
	at org.terasology.world.block.entity.BlockCommands.giveBlock(BlockCommands.java:221) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_45]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_45]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_45]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_45]
	at org.terasology.logic.console.commandSystem.AbstractCommand.execute(AbstractCommand.java:237) ~[classes/:na]
	... 25 common frames omitted
23:08:15.772 [main] INFO  o.t.logic.console.ConsoleImpl - [ERROR] An error occurred while executing command 'giveBlock': No block found for 'eastereggdetector'
```


<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/32347690-better-error-message-on-giveblock-command-for-non-existing-blocks?utm_campaign=plugin&utm_content=tracker%2F5279&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F5279&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java
+++ b/engine/src/main/java/org/terasology/world/block/entity/BlockCommands.java
@@ -218,13 +218,30 @@ public class BlockCommands extends BaseComponentSystem {
                 throw new IllegalArgumentException("No block found for '" + uri + "'");
             }
         } else if (matchingUris.isEmpty()) {
-            throw new IllegalArgumentException("No block found for '" + uri + "'");
+        	return suggestItemIfAvailable(uri);
         } else {
             StringBuilder builder = new StringBuilder();
             builder.append("Non-unique block name, possible matches: ");
             Joiner.on(", ").appendTo(builder, matchingUris);
             return builder.toString();
         }
+    }
+    
+    /**
+     * Tells players that their request matched no blocks, and directs them to giveItem if an item matches.
+     * 
+     * @param uri the URI to use to look for an item
+     */
+    private String suggestItemIfAvailable(String uri){
+    	Set<ResourceUrn> matchingItems = assetManager.resolve(uri, Prefab.class);
+    	StringBuilder result = new StringBuilder();
+    	result.append("No block found for "+uri);
+    	if(matchingItems.size() != 0){
+    		result.append(". ");
+    		result.append("Item matches found, use 'giveItem' to request one: ");
+    		Joiner.on(", ").appendTo(result, matchingItems);
+    	}
+    	return result.toString();
     }
 
     /**

